smarts/core/lanepoints.py

```diff
@@ -527,9 +519,10 @@ def _closest_linked_lp_in_kd_tree_batched(
         points, linked_lps, tree: KDTree, k: int = 1
     ):
         p2ds = np.array([vec_2d(p) for p in points])
-        closest_indices = tree.query(
-            p2ds, k=min(k, len(linked_lps)), return_distance=False, sort_results=True
+        _, closest_indices = tree.query(
```

Comment:
![image](https://user-images.githubusercontent.com/47334253/157322307-36425e3b-d42a-43e9-a60e-6d4ae994af01.png)

The distance return by this method is deprecated and is removed in the latest version of `scipy`. We should be careful here whether we want to upgrade to latest version of scipy to avoid the distance return or lock the version if we have a dependency requirement.

Reply:
We could also conditionally unpack. Although that might be un-optimal. It might also be possible to determine that condition once by wrapping the query method.

Reply:
Only the condition where k=None which is deprecated. This doesn't affect our use so I don't think we need to worry too much about locking the dependency version. See https://docs.scipy.org/doc/scipy-1.5.4/reference/generated/scipy.spatial.KDTree.query.html for an older version